invenio_rdm_records/services/secret_links/service.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM Secret Link Service."""
+
+from datetime import datetime
+
+import arrow
+from flask_babelex import lazy_gettext as _
+from invenio_db import db
+from invenio_drafts_resources.services.records import RecordService
+from invenio_records_resources.services.records.schema import \
+    ServiceSchemaWrapper
+from marshmallow.exceptions import ValidationError
+
+
+class SecretLinkService(RecordService):
+    """RDM Secret Link service."""
+
+    def __init__(self, config, secret_links_service=None):
+        """Constructor for Secret Link Service."""
+        super().__init__(config)
+        self._secret_links = secret_links_service
```

Comment:
major: I think there is a conceptual error...  The *secret_links* service should be "independent" like e.g. [`FileService`](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/files/service.py#L22) and inherit from `Service`. Then, it would be `RDMRecordsService` who would have the `secret_links` attribute and property (i.e. `RDMRecordsService` would do something similar to [`RecordsService`](https://github.com/inveniosoftware/invenio-drafts-resources/blob/master/invenio_drafts_resources/services/records/service.py#L31)...

All this said, maybe I'm missing a reason why it has to be like these 😅